text/0136-soft-deleted-documents.md

```diff
@@ -0,0 +1,39 @@
+# Soft Deleted Documents
+
+## 1. Summary
+
+This specification describes the internals of the soft-deleted documents algorithm.
+
+## 2. Motivation
+
+Deleting documents is extremely slow and can happen when;
+- You delete a single document.
+- You delete a batch of documents.
+- You update one or multiple documents (i.e., the primary key is the same, but the document's content is not the same).
+
+The purpose of the soft-deleted documents feature is to make the deletion of documents almost instantaneous by **not** deleting the document when asked.
+
+## 3. Functional Specification
+
+Instead of deleting the documents, we mark them internally as deleted and then exclude them from all the other algorithms of the engine.
+That's fast but takes space; thus, at some point, we need to _really_ delete the soft deleted documents.
+This can happen for two reasons;
+- When 90% of the total available space is used.
+- When 10% of the total space is dedicated to the soft deleted documents.
+
+The idea is good, but there are two little problems;
+1. We don't know the size a document really occupies.
+  This means we don't know the size used by the soft deleted documents.
+  That can be imprecise in the case of a really heterogeneous dataset with large and small documents.
+2. We don't know the total available space. The only information available to meilisearch is the `max-index-size` which is by default at 100GB, but meilisearch could be deployed on a smaller disk.
```

Comment:
I see now why it's important to set the `max-index-size`, even for disks < 100GB. Thanks @irevoire!